tests/integration_tests/performance/test_benchmarks.py

```diff
@@ -34,51 +49,44 @@ def cargo_bench():
             if executable:
                 executables.append(executable)
 
-    output = ""
-
     for executable in executables:
-        output += utils.run_cmd(
-            f"CARGO_TARGET_DIR=../build/cargo_target taskset -c 1 {executable} --bench"
-        ).stdout
+        utils.run_cmd(
+            f"CARGO_TARGET_DIR=build/cargo_target taskset -c 1 {executable} --bench --save-baseline {baseline_name}"
+        )
 
-    return output
+    return Path(os.getcwd()) / "build" / "cargo_target" / "criterion"
 
 
-@pytest.mark.no_block_pr
-@pytest.mark.timeout(600)
-def test_no_regression_relative_to_target_branch():
-    """
-    Run the microbenchmarks in this repository, comparing results from pull
-    request target branch against what's achieved on HEAD
-    """
-    # First, run benchmarks on pull request target branch (usually main). For this, cache the commit at which
-    # the test was originally executed
-    _, pr_head_commit_sha, _ = utils.run_cmd("git rev-parse HEAD")
-    utils.run_cmd(f"git switch {TARGET_BRANCH}")
-    cargo_bench()
-
-    # Switch back to pull request, and run benchmarks again. Criterion will automatically notice that
-    # data from a previous run exists, and do a comparison
-    utils.run_cmd(f"git checkout {pr_head_commit_sha}")
-    criterion_output = cargo_bench()
-
-    # Criterion separates reports for benchmarks by two newlines. We filter and print the ones
-    # that contain the string 'Performance has regression.', which criterion uses to indicate a regression
-    regressions_only = "\n\n".join(
-        result
-        for result in criterion_output.split("\n\n")
-        if "Performance has regressed." in result
-    )
-
-    for benchmark in os.listdir("../build/cargo_target/criterion"):
+def compare_results(location_a_baselines: Path, location_b_baselines: Path):
+    """Compares the two recorded criterion baselines for regressions, assuming that "A" is the baseline from main"""
+    for benchmark in location_b_baselines.glob("*"):
         with open(
-            f"../build/cargo_target/criterion/{benchmark}/new/estimates.json",
-            encoding="utf-8",
+            benchmark / "b_baseline" / "estimates.json", encoding="utf-8"
         ) as file:
             data = json.load(file)
```

Comment:
```python
json.load(file.open(encoding...))
```

Reply:
Ah, right, I did this somewhere else and then promtly forgot about it for this commit :joy: